.github/workflows/build-scheds.yml

```diff
@@ -48,11 +48,14 @@ jobs:
       # Build a minimal kernel (with sched-ext enabled) using virtme-ng
       - run: cd linux && vng -v --build --config .github/workflows/sched-ext.config
 
+      # Generate kernel headers
+      - run: cd linux && make headers
+
       ### END DEPENDENCIES ###
 
       # The actual build:
-      - run: meson setup build -Dkernel=$(pwd)/linux
-      - run: meson compile -C build
+      - run: meson setup build -Dkernel=$(pwd)/linux -Dkernel_headers=./linux/usr/include
+      - run: meson compile -C build --jobs=1
```

Comment:
Just out of curiosity, why do we want to avoid overloading the system during a CI run given that all it's doing is building + testing?

Reply:
At the moment this is not critical, but in the future, if we add more Rust components, they'll be all built in parallel by meson in a N^2 way (where N is the amount of CPUs), so there's the risk that at some point we may hit OOM conditions.

The issue is that meson will try to parallelize the sub-projects spawning N tasks, each task will run cargo that is not aware of meson, so it'll also try to spawn N tasks, causing a massive NxN parallel build workload... serializing each cargo run would still build the Rust components in parallel, but limited to N tasks.